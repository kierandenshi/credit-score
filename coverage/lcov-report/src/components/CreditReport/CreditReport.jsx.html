<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/CreditReport/CreditReport.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/CreditReport</a> CreditReport.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
    fetchReportBegin,
    getReportData,
} from '@redux/report';
import SlideShow from '@components/SlideShow';
import CreditRating from '@components/CreditRating';
import LongTermDebt from '@components/LongTermDebt';
import './_CreditReport.scss';
&nbsp;
export class CreditReport extends Component {
    componentDidMount() {
        this.props.fetchReportBegin();
    }
&nbsp;
    renderReportPages() {
        const { data } = this.props;
&nbsp;
        return (
            &lt;div className={'credit-report__report-pages'}&gt;
                &lt;SlideShow&gt;
                    &lt;CreditRating data={data} /&gt;
                    &lt;LongTermDebt data={data} /&gt;
                &lt;/SlideShow&gt;
            &lt;/div&gt;
        );
    }
&nbsp;
    renderReportLoading() {
        return (
            &lt;div className={'credit-report__loading-spinner'}&gt;Loading&lt;/div&gt;
        );
    }
&nbsp;
    render() {
        const { hasData } = this.props;
&nbsp;
        return (
            &lt;div className={'credit-report'}&gt;
                { hasData ? this.renderReportPages() : this.renderReportLoading() }
            &lt;/div&gt;
        );
    }
}
&nbsp;
CreditReport.defaultProps = {
    data: {},
};
&nbsp;
CreditReport.propTypes = {
    fetchReportBegin: PropTypes.func.isRequired,
    hasData: PropTypes.bool.isRequired,
    data: PropTypes.shape({}),
};
&nbsp;
export default connect(
<span class="fstat-no" title="function not covered" >    st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
        hasData: state.report.hasData,
        data: getReportData(state),
    }),
    {
        fetchReportBegin,
    },
)(CreditReport);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jan 16 2018 09:27:06 GMT+0000 (GMT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
