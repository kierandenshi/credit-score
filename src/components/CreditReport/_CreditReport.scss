@import '~@css/includes/helpers';

.credit-report {
  &__report-pages {
    position: relative;
    width: 298px;
    height: 298px;
    margin: auto;
    overflow: hidden;

    &::before {
      content: '';
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      border-radius: 50%;
      background: white;
      opacity: 0.2;
    }

    @include media('>phone') {
      width: 414px;
      height: 414px;
    }
  }
}
