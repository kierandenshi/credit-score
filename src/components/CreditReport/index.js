export { default } from './CreditReport.jsx';
