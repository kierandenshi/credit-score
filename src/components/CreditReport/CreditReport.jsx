import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
    fetchReportBegin,
    getReportData,
} from '@redux/report';
import SlideShow from '@components/SlideShow';
import CreditRating from '@components/CreditRating';
import LongTermDebt from '@components/LongTermDebt';
import './_CreditReport.scss';

export class CreditReport extends Component {
    componentDidMount() {
        this.props.fetchReportBegin();
    }

    renderReportPages() {
        const { data } = this.props;

        return (
            <div className={'credit-report__report-pages'}>
                <SlideShow>
                    <CreditRating data={data} />
                    <LongTermDebt data={data} />
                </SlideShow>
            </div>
        );
    }

    renderReportLoading() {
        return (
            <div className={'credit-report__loading-spinner'}>Loading</div>
        );
    }

    render() {
        const { hasData } = this.props;

        return (
            <div className={'credit-report'}>
                <div className={'credit-report__main-section'}>
                    { hasData ? this.renderReportPages() : this.renderReportLoading() }
                </div>
            </div>
        );
    }
}

CreditReport.defaultProps = {
    data: {},
};

CreditReport.propTypes = {
    fetchReportBegin: PropTypes.func.isRequired,
    hasData: PropTypes.bool.isRequired,
    data: PropTypes.shape({}),
};

export default connect(
    state => ({
        hasData: state.report.hasData,
        data: getReportData(state),
    }),
    {
        fetchReportBegin,
    },
)(CreditReport);
