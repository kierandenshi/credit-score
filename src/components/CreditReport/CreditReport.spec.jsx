import React from 'react';
import { shallow } from 'enzyme';
import { getProps, updateProps } from '@test/helpers';
import { CreditReport } from './CreditReport.jsx';

const initialProps = {
    hasData: false,
    data: {},
    fetchReportBegin: jest.fn(),
    getReportData: jest.fn(),
};

const nextProps = {
    hasData: true,
};

describe('Credit report component', () => {
    const wrapper = shallow(<CreditReport {...getProps(initialProps)} />);

    it('should render', () => {
        expect(wrapper.find('.credit-report')).toMatchSnapshot();
    });

    it('should render a loading message', () => {
        expect(wrapper.find('.credit-report__loading-spinner')).toHaveLength(1);
        expect(wrapper.find('.credit-report__report-pages')).toHaveLength(0);
    });

    it('should render the report pages', () => {
        updateProps(wrapper, initialProps, nextProps);
        expect(wrapper.find('.credit-report__loading-spinner')).toHaveLength(0);
        expect(wrapper.find('.credit-report__report-pages')).toHaveLength(1);
    });
});
