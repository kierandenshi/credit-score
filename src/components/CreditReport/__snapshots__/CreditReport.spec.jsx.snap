// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit report component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <CreditReport
      data={Object {}}
      fetchReportBegin={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      getReportData={[MockFunction]}
      hasData={false}
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="credit-report__loading-spinner"
        >
          Loading
        </div>,
        "className": "credit-report",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading",
          "className": "credit-report__loading-spinner",
        },
        "ref": null,
        "rendered": "Loading",
        "type": "div",
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="credit-report__loading-spinner"
          >
            Loading
          </div>,
          "className": "credit-report",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading",
            "className": "credit-report__loading-spinner",
          },
          "ref": null,
          "rendered": "Loading",
          "type": "div",
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="credit-report__loading-spinner"
      >
        Loading
      </div>,
      "className": "credit-report",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading",
        "className": "credit-report__loading-spinner",
      },
      "ref": null,
      "rendered": "Loading",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="credit-report__loading-spinner"
        >
          Loading
        </div>,
        "className": "credit-report",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading",
          "className": "credit-report__loading-spinner",
        },
        "ref": null,
        "rendered": "Loading",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
